cg_lims/EPPs/qc/sequencing_quality_checker.py

```
@@ -0,0 +1,87 @@
+import logging
+import sys
+from typing import List
+
+from cg_lims.EPPs.qc.sequencing_artifact_manager import SequencingArtifactManager
+from cg_lims.exceptions import LimsError
+from cg_lims.models.sequencing_metrics import SampleLaneSequencingMetrics
+from cg_lims.status_db_api import StatusDBAPI
+
+LOG = logging.getLogger(__name__)
+
+
+class SequencingQualityChecker:
+    READS_MIN_THRESHOLD = 1000
+
+    def __init__(
+        self, status_db_api: StatusDBAPI, artifact_manager: SequencingArtifactManager
+    ) -> None:
+        self.artifact_manager: SequencingArtifactManager = artifact_manager
+        self.status_db_api: StatusDBAPI = status_db_api
+
+        self.q30_threshold: int = self.artifact_manager.q30_threshold
+        self.flow_cell_name: str = self.artifact_manager.flow_cell_name
+
+        self.samples_not_passing_qc_count: int = 0
+
+    def _get_sequencing_metrics(self) -> List[SampleLaneSequencingMetrics]:
+        try:
+            return self.status_db_api.get_sequencing_metrics_for_flow_cell(
+                self.flow_cell_name
+            )
+        except LimsError as e:
+            error_message: str = (
+                f"Could not retrieve sequencing metrics for {self.flow_cell_name}: {e}"
+            )
+            LOG.error(error_message)
+            sys.exit(error_message)
```

Not a good pattern, makes it difficult to understand the flow. Better to raise an exception

---

https://github.com/Clinical-Genomics/cg_lims/pull/416/commits/cecbcc9955b43128bc55c89dd9eefc3ef527c6f4